app/helpers/argo_helper.rb

```diff
@@ -179,7 +179,7 @@ def render_foxml_link(document, link_text = 'FoXML', opts = {:target => '_blank'
   end
 
   def render_solr_link(document)
-    return '' unless current_user.is_admin
+    return '' unless can? :read, document
```

Comment:
@LynnMcRae cancan provides the ability to separate the "who" from the "what". The view asks "who can read (view) this document", and the ability class says "admins can read (view) documents).